### PR TITLE
Bump h2 from 1.4.197 to 2.2.220

### PR DESCRIPTION
**Pull request by tkobayas**
- remove MVCC

**referenced Pull Requests**: 

* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2356
* https://github.com/kiegroup/drools/pull/5395
* https://github.com/kiegroup/droolsjbpm-integration/pull/2983
* https://github.com/kiegroup/jbpm-work-items/pull/285
* https://github.com/kiegroup/kie-jpmml-integration/pull/83

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Removing `;MVCC=true` and adding `;MODE=LEGACY;OLD_INFORMATION_SCHEMA=TRUE` is not enough.

```
[2023-07-20T03:28:27.827Z] 2023-07-19 23:28:26,670 [main] [SqlExceptionHelper] ERROR Table "CORRELATIONPROPERTYINFO" not found; SQL statement:
```

Even table creation seems to be failing.

It is out of my expertise. @gmunozfe  Could you kindly help on this or involve someone suitable? Thanks!

**Comment by tkobayas** [comment]
A great number of test failures are resolved. Now 7 failures. Thank you and please check them! @gmunozfe 
```
Test Result (7 failures / +7)
    org.jbpm.test.container.test.ejbservices.process.EMigrateProcessTest.testUpgradeProcessInstancesWithNodeMapping
    org.jbpm.test.container.test.ejbservices.process.EMigrateProcessTest.testUpgradeProcessInstance
    org.jbpm.test.container.test.ejbservices.process.EMigrateProcessTest.testUpgradeProcessInstances
    org.jbpm.test.container.test.ejbservices.process.EMigrateProcessTest.testUpgradeProcessInstanceWithNodeMapping
    org.jbpm.test.container.test.ejbservices.task.EGetTaskTest.testGetTaskInstanceInfo
    org.jbpm.test.container.test.ejbservices.task.EGetTaskTest.testTaskQuery
    org.jbpm.test.container.test.ejbservices.task.ETaskOperationTest.testStartAndComplete
```

**Comment by tkobayas** [comment]
0 failed. Thanks! @gmunozfe 

Now I re-triggered 
- PR test in https://github.com/kiegroup/droolsjbpm-integration/pull/2983
- FDB in https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2356

**Comment by gmunozfe** [comment]
@tkobayas Great, however, I'm not very convinced with my last commit (it's a workaround valid for h2 to skip the hibernate bug tracked as HHH-13711, but won't work for the rest of the databases). The point here is these container tests are using wildfly-dist 23.0.0-Final, which doesn't include the hibernate-core fix (from 5.4.13), so dropping tables for cleaning up will only work if we replace that module or apply the workaround, wdyt?

**Comment by tkobayas** [comment]
@gmunozfe Thanks, it looks good.

**Comment by tkobayas** [comment]
GHA Java11 : Not related to this PR
```
2023-07-25T06:20:50.2312963Z [ERROR] Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.23 s <<< FAILURE! - in org.jbpm.process.workitem.email.SendHtmlTest
2023-07-25T06:20:50.2313761Z [ERROR] sendHtmlWithBadAuthentication(org.jbpm.process.workitem.email.SendHtmlTest)  Time elapsed: 0.006 s  <<< ERROR!
2023-07-25T06:20:50.2314444Z java.lang.RuntimeException: java.net.BindException: Address already in use (Bind failed)
```

**Review by gmunozfe — APPROVED** [review]
Looks good to me, thanks a lot @tkobayas for taking care of this!

**Review by danielezonca — APPROVED** [review]
Overall LGTM, some minor comments, feel free to address them with a following PR if you prefer

File: jbpm-xes/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
   
   <properties>
     <version.commons-cli>1.4</version.commons-cli>
-    <version.com.h2database>1.4.193</version.com.h2database>
+    <version.com.h2database>2.2.220</version.com.h2database>
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-xes/pom.xml:33]: Right, that version override is not needed, removing it...

File: jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/JbpmDialectResolver.java [hunk 2]
```diff
@@ -34,7 +34,8 @@ public Dialect resolveDialect(DialectResolutionInfo info){
         if ("Adaptive Server Enterprise".equals(info.getDatabaseName())) {
            return new SybaseJbpmDialect(); 
         } else {
-            d = StandardDialectResolver.INSTANCE.resolveDialect(info);
+            DialectResolver dialectResolver = new StandardDialectResolver();
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/JbpmDialectResolver.java:37]: Yes, that constant was removed in 5.4.0, https://github.com/hibernate/hibernate-orm/commit/c108f48674e87bd5577468992b65c7afc877e418

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ActivityTest.java [hunk 3]
```diff
@@ -90,6 +91,7 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.jbpm.test.persistence.util.PersistenceUtil.cleanUp;
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ActivityTest.java:94]: removed

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ErrorEventTest.java [hunk 4]
```diff
@@ -35,6 +35,7 @@
 import org.jbpm.workflow.instance.WorkflowProcessInstance;
 import org.jbpm.workflow.instance.WorkflowRuntimeException;
 import org.junit.After;
+import org.junit.Before;
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ErrorEventTest.java:38]: Removed

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/JbpmBpmn2TestCase.java [hunk 5]
```diff
@@ -106,6 +106,7 @@
 import org.slf4j.LoggerFactory;
 import org.xml.sax.SAXException;
 
+import static org.jbpm.test.persistence.util.PersistenceUtil.cleanUp;
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/JbpmBpmn2TestCase.java:109]: Removed

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ActivityTest.java [hunk 6]
```diff
@@ -486,7 +489,8 @@ public void testUserTaskWithDataStoreScenario() throws Exception {
         StatefulKnowledgeSession ksession = createKnowledgeSession(kbase);
         ksession.getWorkItemManager().registerWorkItemHandler("Human Task",
                 new DoNothingWorkItemHandler());
-        ksession.startProcess("UserProcess");
+        ProcessInstance pi = ksession.startProcess("UserProcess");
+        ksession.abortProcessInstance(pi.getId());
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ActivityTest.java:493]: Rolledback

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 7]
```diff
@@ -774,21 +787,25 @@ public void afterProcessStarted(ProcessStartedEvent event) {
         });
         ksession.signalEvent("MySignal", "NewValue");
         assertThat(list.size()).isEqualTo(1);
+        try {
+          ksession.abortProcessInstance(list.get(0));
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:791]: Rolledback (this was not intended but kept after working on it when looking for the reason of not cleaning up)

File: jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 8]
```diff
@@ -797,12 +814,16 @@ public void afterProcessStarted(ProcessStartedEvent event) {
         });
         ksession.signalEvent("Message-HelloMessage", "NewValue");
         assertThat(list.size()).isEqualTo(1);
+        try {
+          ksession.abortProcessInstance(list.get(0));
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:818]: Rolledback

**Comment by gmunozfe** [comment]
@danielezonca @tkobayas I've proposed https://github.com/tkobayas/jbpm/pull/4 covering all the changes coming from the review

**Comment by tkobayas** [comment]
@gmunozfe Thank you!

**Comment by tkobayas** [comment]
The changes are only test cases and jbpm-xes which doesn't have downstream projects, so no need to run FDB in droolsjbpm-build-bootstrap again.

**Comment by tkobayas** [comment]
0 failed :)

**Pull request merged**